packages/example/pages/index.tsx

```diff
@@ -1,109 +1,153 @@
-import { useWallet, useWalletManager } from '@cosmos-kit/react'
-import { CosmosWalletStatus } from '@cosmos-kit/types'
-import type { NextPage } from 'next'
-import React, { useCallback, useState } from 'react'
+import { useWallet } from "@cosmos-kit/react";
+import { chainInfos } from "../config";
+import { Box, Center, Grid, GridItem, Icon, Stack, useColorModeValue } from "@chakra-ui/react";
+import { MouseEventHandler } from "react";
+import { FiAlertTriangle } from "react-icons/fi";
+import { Astronaut, Error, ChainOption, ChooseChain, Connected, ConnectedShowAddress, ConnectedUserInfo, Connecting, ConnectStatusWarn, CopyAddressBtn, Disconnected, handleSelectChainDropdown, NotExist, Rejected, RejectedWarn, WalletConnectComponent } from "../components";
+import { getWalletPrettyName } from "@cosmos-kit/registry";
 
-const Home: NextPage = () => {
-  const { connect, disconnect } = useWalletManager()
-  const {
-    status: walletStatus,
-    error,
-    name,
-    address,
-    signingCosmWasmClient,
-  } = useWallet()
+const Home = () => {
+  const walletManager = useWallet();
+  const { connect, disconnect, openModal, setCurrentChain,
+    walletStatus, username, address, message,
+    currentChainName: chainName, currentWalletName } = walletManager;
 
-  const [contractAddress, setContractAddress] = useState('')
-  const [msg, setMsg] = useState('')
-  const [status, setStatus] = useState('')
+  // Events
+  const onClickConnect: MouseEventHandler = async (e) => {
+    e.preventDefault();
+    openModal();
+    await connect();
+  };
 
-  const execute = useCallback(async () => {
-    if (!address || !signingCosmWasmClient) return
+  const onClickDisconnect: MouseEventHandler = async (e) => {
+    e.preventDefault();
+    openModal();
```

Comment:
is this how we control the disconnect behavior for open/close view?

also, wondering if we should name this `openView` and `closeView` instead of modal so we can support the dropdown and modal?

Reply:
yes, we can control the disconnect behavior for open/close view by `onClickDisconnect `, similar with `reconnect` and `connect`

renaming to `openView` and `closeView` is done. also for other necessary `modal` keywords